Allow library to parse ISO 8601 dates, in addition to existing one.

noordawod opened the pull request:
https://en.wikipedia.org/wiki/ISO_8601

The library will now support both ISO 8601 and RFC 822 dates.

noordawod committed:
Allow library to parse ISO 8601 dates, in addition to existing one.

einmalfel commented:
thanks for contribution, @noordawod ! PR looks cool, will merge it soon. Could you please provide a link to feed that uses ISO 8601 date formatting, so I'll be able to make a test for this feature?

noordawod commented:
Right on it, @einmalfel. Since I mostly trusted my judgement on the matter, let me perform this test myself and submit any more patches (if needed) and then I'll send you a feed.

noordawod committed:
One date parsing function to rule them all!
Use Utils.parseDate() to parse any supprted date format, even when developers don't follow standards (unfortunately).

noordawod commented:
@einmalfel 

Okay, now I've made sure the patch works and even widened the coverage of `Utils.parseDate()` so that it's the only function being used to parse dates. Why? Well, because developers don't follow standards unfortunately and we have to try few formats to parse the date before giving up...

Anyway, now the patch is complete and tested against few feeds.

Here's one for testing: http://feeds.feedburner.com/AniaMaluje-BlogUrodowy

By the way, we're using your library to cover our Atom/RSS needs in an upcoming mobile app we're working on, so I'll probably submit more pulls as time goes by. Thanks for the great work :smile:

noordawod commented:
@einmalfel 

You think you could take a look at the two pulls today? Would be nice :smile: 

Thanks!

einmalfel commented:
@noordawod 
Sorry about the delay! I'll check & merge PRs in the evening. I wont have a chance to look into new code until then :(

noordawod commented:
Sounds good, @einmalfel. I tested both obviously but would be nice to have another set of eyes take a look.

einmalfel commented on earl/src/main/java/com/einmalfel/earl/Utils.java:
```diff
@@ -29,17 +29,88 @@
   }
 
+  @Nullable
+  private static Date parseRFC8601Date(@NonNull String dateString) {
```
If I understand correctly, it's ISO 8601, not RFC

noordawod replied:
Correct. Did you change the code or should I do it?

einmalfel commented:
I'll change it

On Thu, Oct 27, 2016, 12:35 Noor Dawod notifications@github.com wrote:

> ## _@noordawod_ commented on this pull request.
> 
> In earl/src/main/java/com/einmalfel/earl/Utils.java
> https://github.com/einmalfel/Earl/pull/12:
> 
> > ```
> >    return null;
> >  }
> > ```
> > 
> >    }
> -  @Nullable
> -  private static Date parseRFC8601Date(@NonNull String dateString) {
> 
> Correct. Did you change the code or should I do it?
> 
> —
> You are receiving this because you modified the open/close state.
> 
> Reply to this email directly, view it on GitHub
> https://github.com/einmalfel/Earl/pull/12, or mute the thread
> https://github.com/notifications/unsubscribe-auth/AChLrbeE3Rn11ZclEqov_CneeLUYMvzIks5q4HBOgaJpZM4KfFnW
> .
